apiserver/common/crossmodel/auth.go

```diff
@@ -147,7 +147,7 @@ func (a *AuthContext) CheckOfferAccessCaveat(caveat string) (*offerPermissionChe
 func (a *AuthContext) CheckLocalAccessRequest(details *offerPermissionCheck) ([]checkers.Caveat, error) {
 	authlogger.Debugf("authenticate local offer access: %+v", details)
 	if err := a.checkOfferAccess(a.systemState.UserPermission, details.User, details.OfferUUID); err != nil {
-		return nil, errors.Trace(err)
+		return nil, errors.Annotatef(err, "offer access check for user %q to offer %q in model %q", details.User, details.OfferUUID, details.SourceModelUUID)
```

Comment:
As a general rule of thumb, we try not to return any information about the user details when performing authn/authz. Juju has a policy of returning just `ErrPerm` (`permission denied`). Once you try to add additional information it gives rise to accidentally leaking details, which could be used for rainbow attacks or similar. Even at the cost of user experience.

In reality the endpoint is using the details sent via the macaroon, so surely the origin can use the original information to supplement the error? If the details come from a third party that isn't the source of the request, then we might be plumb out of luck. The details do get logged, so checking the logs is the workaround.

Reply:
Gotcha, I'll look at whether we can do the logging on the consumer side by using the macaroon.

Reply:
what if you just log this information and return a generic ErrPerm error?

Reply:
We don't want to log it here because it won't show up in the model logs of the consuming model (I think).